### PR TITLE
feat(diagram-converter): conditional events expression handling

### PR DESCRIPTION
**Pull request by venetrius**
related to:
-  camunda/camunda-7-to-8-migration-tooling/issues/682

# Pull Request Template

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Test-only changes (no production code changes)

### Black-Box Testing Requirements
- [ ] Tests follow **black-box testing approach**: verify behavior through observable outputs (logs, C8 API queries, real-world skip scenarios)
- [ ] Tests **DO NOT** access implementation details (`DbClient`, `IdKeyMapper`, `..impl..` packages except logging constants)
- [ ] Architecture tests pass (`ArchitectureTest` validates these rules)

### Test Coverage
- [ ] Added tests for new functionality
- [ ] Updated tests for modified functionality
- [ ] All tests pass locally

## Architecture Compliance

Run architecture tests to ensure compliance:
```bash
mvn test -Dtest=ArchitectureTest
```

If architecture tests fail, refactor your tests to use:
- `LogCapturer` for log assertions
- `camundaClient.new*SearchRequest()` for C8 queries
- Real-World skip scenarios (e.g., migrate children without parents)

## Documentation
- [ ] Updated TESTING_GUIDELINES.md if adding new test patterns
- [ ] Added Javadoc comments for public APIs
- [ ] Updated README if user-facing changes

## Checklist
- [ ] Code follows project style guidelines
- [ ] Self-reviewed the code
- [ ] Added comments for complex logic
- [ ] No new compiler warnings
- [ ] Dependent changes have been merged

### CODE REVIEW

**Comment by venetrius** [comment]
This PR covers the following: 
```
### PR 2: Expression transformation
- Transform JUEL expressions to FEEL using `ExpressionTransformer`
- Handle FEEL expressions (add `=` prefix if missing)
- Handle script languages (Groovy, JavaScript) - add TASK message for manual conversion
- Handle external scripts (`camunda:resource`) - add TASK message
- Add unit tests for expression scenarios
```

Handling`Edge cases` is going to go into a new PR to iterate quickly. See the issue for details: https://github.com/camunda/camunda-7-to-8-migration-tooling/issues/682 .

To help with testing I have added a [QA note](https://github.com/camunda/camunda-7-to-8-migration-tooling/issues/682#issuecomment-3852754157).

File: diagram-converter/core/src/main/java/io/camunda/migration/diagram/converter/visitor/impl/ConditionVisitor.java [hunk 1]
```diff
@@ -63,18 +63,47 @@ private boolean isInsideConditionalEventDefinition(DomElementVisitorContext cont
   @Override
   protected void visitBpmnElement(DomElementVisitorContext context) {
     String expression = context.getElement().getTextContent();
-    if (expression == null || expression.isBlank()) {
```
[the remark this answers is not in the document]
Comment (venetrius) [reply diagram-converter/core/src/main/java/io/camunda/migration/diagram/converter/visitor/impl/ConditionVisitor.java]: No I is not necessary . I added support for handling BPMNs with external scripts. When using external scripts the expression is null.
